compiler/ETA/CodeGen/Expr.hs

```diff
@@ -319,3 +341,9 @@ cgAlgAltRhss binder alts = do
       allBranches = [ (getDataConTag con, code)
                     | (DataAlt con, code) <- taggedBranches ]
   return (maybeDefault, branches)
+
+cgTick :: Tickish Id -> CodeGen ()
+cgTick (SourceNote srcSpan _) = do
+  let srcLoc = realSrcSpanStart srcSpan
+  addLineNumber $ srcLocLine srcLoc
```

Comment:
I just realised there's a tricky corner case here. You should only addLineNumber if the file corresponds to the current module, otherwise you'll get a line number that doesn't match the source file.

Reply:
Ok, just out of curiosity, do you know the cause of that case?

Reply:
Here's an example:

`GHC.List`
```
creating new closure...errorEmptyList
StgTick errorEmptyList <GHC/List.hs:(1010,1)-(1011,48)>
StgTick build <GHC/Base.hs:785:1-18>
StgLet sat_sNKV
```

In this example, the line number would show up as 785 with source file as `GHC/List.hs` when it should be 1010.

Reply:
Aja, i see, it generates ticks for all the call stack, including external modules ones